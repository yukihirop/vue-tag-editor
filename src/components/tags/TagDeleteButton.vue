<template>
  <span>
    <button @click="emitDeleteTag">
      x
    </button>
  </span>
</template>

<script>
export default {
  name: "TagDeleteButton",
  props:{
    tagname: {
      type: String,
      default: ''
    },
    eventHub: {
      type: Object,
      default(){
        return null
      }
    }
  },
  methods: {
    emitDeleteTag() {
      this.$emit('delete-tag')
      this.eventHub.$emit('delete-tag', this.tagname)
    }
  }
}
</script>