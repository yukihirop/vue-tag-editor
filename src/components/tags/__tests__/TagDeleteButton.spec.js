import { createLocalVue, mount } from '@vue/test-utils'
import TagDeleteButton from '../TagDeleteButton'

const localVue = createLocalVue()

const wrapperEventHub = mount(TagDeleteButton, {
  localVue,
  propsData: {
    tagname: 'test'
  }
})
const eventHub = wrapperEventHub.vm

describe('TagDeleteButton', () => {
  const wrapper = mount(TagDeleteButton, {
    localVue,
    propsData: {
      tagname: 'test',
      eventHub: eventHub
    }
  })

  it('correct name options', () => {
    expect(wrapper.vm.$options.name).toBe('TagDeleteButton')
  })

  it('renders correctly (snapshots)', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders the correct markup', () => {
    expect(wrapper.html()).toContain("x")
  })

  it('has a delete button', () => {
    expect(wrapper.contains('button')).toBe(true)
  })

  it('delete button click should call emitDeleteTag', () => {
    const button = wrapper.find('button')
    button.trigger('click')

    expect(wrapper.emitted('delete-tag')).toBeTruthy()
    expect(wrapper.emitted('delete-tag').length).toBe(1)
    expect(wrapperEventHub.emitted('delete-tag')).toBeTruthy()
    expect(wrapperEventHub.emitted('delete-tag').length).toBe(1)
    expect(wrapperEventHub.emitted('delete-tag')[0]).toEqual(['test'])
  })
})