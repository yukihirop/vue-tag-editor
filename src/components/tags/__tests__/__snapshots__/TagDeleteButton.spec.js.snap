// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagDeleteButton renders correctly (snapshots) 1`] = `
<span>
  <button
    class="deleteAreaClass"
  >
    <span
      class="deleteContentClass"
    >
      x
    </span>
  </button>
</span>
`;
